azurelinuxagent/distro/default/env.py

```diff
@@ -94,6 +100,22 @@ def handle_dhclient_restart(self):
            self.distro.dhcp_handler.conf_routes()
            self.dhcpid = newpid
 
+     def handle_fstab_update(self):
+        # Only try to mount if the fstab has been modified. This avoids background mounts when admins are working on a unmounted partition
+        # If we find an error, backoff and only check every minute to give time to correct it and avoid filling the log
+        fstabCurrentModifiedStamp = shellutil.run_get_output("stat /etc/fstab -c %Y")
+        if (fstabCurrentModifiedStamp != self.fstabModifiedStamp) and (datetime.datetime.now() > self.lastNotice + datetime.timedelta(seconds=60)):
+            ret, output = shellutil.run_get_output("mount -av")
+            if ret != 0:
+                # Notify the logged on users who just modified the file to take action, immediately and then periodically thereafter...
+                notice =  "[AZURE AGENT] Current fstab settings are invalid.  Please correct the /etc/fstab file content before rebooting using the error information in /var/log/waagent.log"
```

Comment:
I believe that run_get_output() will log to waagent.log when it sees a non-zero return code.  However, I saw at least one instance in testing it where I didn't receive the logging I expected.  I will add an explicit call to logger.error.  thanks.